{"kind":"symbol","sections":[],"identifier":{"url":"doc:\/\/SwiftErrorKit\/documentation\/SwiftErrorKit\/SourceErrorAttribute","interfaceLanguage":"swift"},"hierarchy":{"paths":[["doc:\/\/SwiftErrorKit\/documentation\/SwiftErrorKit"]]},"abstract":[{"type":"text","text":"An error attribute to include information about the high-level component"},{"text":" ","type":"text"},{"type":"text","text":"in which the error emerged."}],"schemaVersion":{"patch":0,"minor":3,"major":0},"topicSections":[{"identifiers":["doc:\/\/SwiftErrorKit\/documentation\/SwiftErrorKit\/SourceErrorAttribute\/external","doc:\/\/SwiftErrorKit\/documentation\/SwiftErrorKit\/SourceErrorAttribute\/input","doc:\/\/SwiftErrorKit\/documentation\/SwiftErrorKit\/SourceErrorAttribute\/internal","doc:\/\/SwiftErrorKit\/documentation\/SwiftErrorKit\/SourceErrorAttribute\/io","doc:\/\/SwiftErrorKit\/documentation\/SwiftErrorKit\/SourceErrorAttribute\/system"],"title":"Enumeration Cases"},{"title":"Initializers","identifiers":["doc:\/\/SwiftErrorKit\/documentation\/SwiftErrorKit\/SourceErrorAttribute\/init(rawValue:)"]},{"title":"Instance Properties","identifiers":["doc:\/\/SwiftErrorKit\/documentation\/SwiftErrorKit\/SourceErrorAttribute\/typeDescription","doc:\/\/SwiftErrorKit\/documentation\/SwiftErrorKit\/SourceErrorAttribute\/valueDescription"]},{"identifiers":["doc:\/\/SwiftErrorKit\/documentation\/SwiftErrorKit\/SourceErrorAttribute\/Equatable-Implementations","doc:\/\/SwiftErrorKit\/documentation\/SwiftErrorKit\/SourceErrorAttribute\/ErrorAttribute-Implementations","doc:\/\/SwiftErrorKit\/documentation\/SwiftErrorKit\/SourceErrorAttribute\/RawRepresentable-Implementations"],"generated":true,"title":"Default Implementations"}],"metadata":{"title":"SourceErrorAttribute","roleHeading":"Enumeration","navigatorTitle":[{"kind":"identifier","text":"SourceErrorAttribute"}],"symbolKind":"enum","externalID":"s:13SwiftErrorKit06SourceB9AttributeO","role":"symbol","modules":[{"name":"SwiftErrorKit"}],"fragments":[{"kind":"keyword","text":"enum"},{"kind":"text","text":" "},{"kind":"identifier","text":"SourceErrorAttribute"}]},"seeAlsoSections":[{"title":"Error attributes in the package","generated":true,"identifiers":["doc:\/\/SwiftErrorKit\/documentation\/SwiftErrorKit\/DebugInfoErrorAttribute","doc:\/\/SwiftErrorKit\/documentation\/SwiftErrorKit\/LocalizationErrorAttribute","doc:\/\/SwiftErrorKit\/documentation\/SwiftErrorKit\/RankErrorAttribute","doc:\/\/SwiftErrorKit\/documentation\/SwiftErrorKit\/SeverityErrorAttribute","doc:\/\/SwiftErrorKit\/documentation\/SwiftErrorKit\/TagErrorAttribute","doc:\/\/SwiftErrorKit\/documentation\/SwiftErrorKit\/TraceErrorAttribute","doc:\/\/SwiftErrorKit\/documentation\/SwiftErrorKit\/UserLevelErrorAttribute"]}],"primaryContentSections":[{"kind":"declarations","declarations":[{"tokens":[{"kind":"keyword","text":"enum"},{"text":" ","kind":"text"},{"kind":"identifier","text":"SourceErrorAttribute"}],"platforms":["macOS"],"languages":["swift"]}]}],"relationshipsSections":[{"title":"Conforms To","kind":"relationships","identifiers":["doc:\/\/SwiftErrorKit\/documentation\/SwiftErrorKit\/ErrorAttribute","doc:\/\/SwiftErrorKit\/SQ","doc:\/\/SwiftErrorKit\/SH","doc:\/\/SwiftErrorKit\/SY"],"type":"conformsTo"}],"variants":[{"paths":["\/documentation\/swifterrorkit\/sourceerrorattribute"],"traits":[{"interfaceLanguage":"swift"}]}],"references":{"doc://SwiftErrorKit/documentation/SwiftErrorKit/SourceErrorAttribute/typeDescription":{"title":"typeDescription","kind":"symbol","identifier":"doc:\/\/SwiftErrorKit\/documentation\/SwiftErrorKit\/SourceErrorAttribute\/typeDescription","fragments":[{"text":"var","kind":"keyword"},{"kind":"text","text":" "},{"text":"typeDescription","kind":"identifier"},{"kind":"text","text":": "},{"text":"String","kind":"typeIdentifier","preciseIdentifier":"s:SS"}],"abstract":[],"type":"topic","role":"symbol","url":"\/documentation\/swifterrorkit\/sourceerrorattribute\/typedescription"},"doc://SwiftErrorKit/documentation/SwiftErrorKit/SourceErrorAttribute/external":{"title":"SourceErrorAttribute.external","kind":"symbol","identifier":"doc:\/\/SwiftErrorKit\/documentation\/SwiftErrorKit\/SourceErrorAttribute\/external","fragments":[{"kind":"keyword","text":"case"},{"text":" ","kind":"text"},{"kind":"identifier","text":"external"}],"abstract":[{"type":"text","text":"“Internal” error in external application."},{"type":"text","text":" "},{"type":"text","text":"Third-party API or service failure, unexpected behavior, …"}],"type":"topic","role":"symbol","url":"\/documentation\/swifterrorkit\/sourceerrorattribute\/external"},"doc://SwiftErrorKit/documentation/SwiftErrorKit/SourceErrorAttribute/io":{"title":"SourceErrorAttribute.io","url":"\/documentation\/swifterrorkit\/sourceerrorattribute\/io","role":"symbol","identifier":"doc:\/\/SwiftErrorKit\/documentation\/SwiftErrorKit\/SourceErrorAttribute\/io","type":"topic","fragments":[{"kind":"keyword","text":"case"},{"kind":"text","text":" "},{"text":"io","kind":"identifier"}],"kind":"symbol","abstract":[{"type":"text","text":"Communication error."},{"type":"text","text":" "},{"type":"text","text":"Network outage, connectivity problem, protocol error, …"}]},"doc://SwiftErrorKit/documentation/SwiftErrorKit/SourceErrorAttribute/RawRepresentable-Implementations":{"type":"topic","url":"\/documentation\/swifterrorkit\/sourceerrorattribute\/rawrepresentable-implementations","kind":"article","identifier":"doc:\/\/SwiftErrorKit\/documentation\/SwiftErrorKit\/SourceErrorAttribute\/RawRepresentable-Implementations","title":"RawRepresentable Implementations","role":"collectionGroup","abstract":[]},"doc://SwiftErrorKit/documentation/SwiftErrorKit/SourceErrorAttribute/Equatable-Implementations":{"title":"Equatable Implementations","kind":"article","identifier":"doc:\/\/SwiftErrorKit\/documentation\/SwiftErrorKit\/SourceErrorAttribute\/Equatable-Implementations","abstract":[],"type":"topic","role":"collectionGroup","url":"\/documentation\/swifterrorkit\/sourceerrorattribute\/equatable-implementations"},"doc://SwiftErrorKit/documentation/SwiftErrorKit/ErrorAttribute":{"fragments":[{"kind":"keyword","text":"protocol"},{"text":" ","kind":"text"},{"kind":"identifier","text":"ErrorAttribute"}],"kind":"symbol","navigatorTitle":[{"text":"ErrorAttribute","kind":"identifier"}],"url":"\/documentation\/swifterrorkit\/errorattribute","role":"symbol","identifier":"doc:\/\/SwiftErrorKit\/documentation\/SwiftErrorKit\/ErrorAttribute","abstract":[{"type":"text","text":"A type that represents an error attribute."}],"title":"ErrorAttribute","type":"topic"},"doc://SwiftErrorKit/documentation/SwiftErrorKit/UserLevelErrorAttribute":{"kind":"symbol","abstract":[{"type":"text","text":"An error attribute to express error level from user perspective."}],"identifier":"doc:\/\/SwiftErrorKit\/documentation\/SwiftErrorKit\/UserLevelErrorAttribute","navigatorTitle":[{"kind":"identifier","text":"UserLevelErrorAttribute"}],"url":"\/documentation\/swifterrorkit\/userlevelerrorattribute","title":"UserLevelErrorAttribute","role":"symbol","fragments":[{"text":"enum","kind":"keyword"},{"text":" ","kind":"text"},{"kind":"identifier","text":"UserLevelErrorAttribute"}],"type":"topic"},"doc://SwiftErrorKit/SH":{"type":"unresolvable","identifier":"doc:\/\/SwiftErrorKit\/SH","title":"Swift.Hashable"},"doc://SwiftErrorKit/SY":{"type":"unresolvable","identifier":"doc:\/\/SwiftErrorKit\/SY","title":"Swift.RawRepresentable"},"doc://SwiftErrorKit/documentation/SwiftErrorKit/LocalizationErrorAttribute":{"type":"topic","kind":"symbol","fragments":[{"kind":"keyword","text":"struct"},{"text":" ","kind":"text"},{"kind":"identifier","text":"LocalizationErrorAttribute"}],"role":"symbol","navigatorTitle":[{"text":"LocalizationErrorAttribute","kind":"identifier"}],"title":"LocalizationErrorAttribute","abstract":[{"text":"An error attribute to provide a localized user message.","type":"text"}],"url":"\/documentation\/swifterrorkit\/localizationerrorattribute","identifier":"doc:\/\/SwiftErrorKit\/documentation\/SwiftErrorKit\/LocalizationErrorAttribute"},"doc://SwiftErrorKit/documentation/SwiftErrorKit/SourceErrorAttribute":{"url":"\/documentation\/swifterrorkit\/sourceerrorattribute","abstract":[{"type":"text","text":"An error attribute to include information about the high-level component"},{"type":"text","text":" "},{"text":"in which the error emerged.","type":"text"}],"navigatorTitle":[{"kind":"identifier","text":"SourceErrorAttribute"}],"type":"topic","kind":"symbol","identifier":"doc:\/\/SwiftErrorKit\/documentation\/SwiftErrorKit\/SourceErrorAttribute","fragments":[{"text":"enum","kind":"keyword"},{"kind":"text","text":" "},{"kind":"identifier","text":"SourceErrorAttribute"}],"title":"SourceErrorAttribute","role":"symbol"},"doc://SwiftErrorKit/documentation/SwiftErrorKit/SourceErrorAttribute/init(rawValue:)":{"title":"init(rawValue:)","kind":"symbol","identifier":"doc:\/\/SwiftErrorKit\/documentation\/SwiftErrorKit\/SourceErrorAttribute\/init(rawValue:)","fragments":[{"text":"init","kind":"identifier"},{"kind":"text","text":"?("},{"text":"rawValue","kind":"externalParam"},{"kind":"text","text":": "},{"preciseIdentifier":"s:SS","text":"String","kind":"typeIdentifier"},{"kind":"text","text":")"}],"abstract":[],"type":"topic","role":"symbol","url":"\/documentation\/swifterrorkit\/sourceerrorattribute\/init(rawvalue:)"},"doc://SwiftErrorKit/documentation/SwiftErrorKit/DebugInfoErrorAttribute":{"kind":"symbol","role":"symbol","fragments":[{"kind":"keyword","text":"struct"},{"text":" ","kind":"text"},{"text":"DebugInfoErrorAttribute","kind":"identifier"}],"navigatorTitle":[{"kind":"identifier","text":"DebugInfoErrorAttribute"}],"abstract":[{"text":"An error attribute to include custom debug info.","type":"text"}],"url":"\/documentation\/swifterrorkit\/debuginfoerrorattribute","identifier":"doc:\/\/SwiftErrorKit\/documentation\/SwiftErrorKit\/DebugInfoErrorAttribute","title":"DebugInfoErrorAttribute","type":"topic"},"doc://SwiftErrorKit/documentation/SwiftErrorKit/SourceErrorAttribute/internal":{"kind":"symbol","url":"\/documentation\/swifterrorkit\/sourceerrorattribute\/internal","type":"topic","identifier":"doc:\/\/SwiftErrorKit\/documentation\/SwiftErrorKit\/SourceErrorAttribute\/internal","fragments":[{"kind":"keyword","text":"case"},{"text":" ","kind":"text"},{"text":"`internal`","kind":"identifier"}],"role":"symbol","abstract":[{"text":"Error in internal application.","type":"text"},{"type":"text","text":" "},{"type":"text","text":"Logic error, bug, or issue with internal operation, …"}],"title":"SourceErrorAttribute.internal"},"doc://SwiftErrorKit/documentation/SwiftErrorKit/TagErrorAttribute":{"kind":"symbol","identifier":"doc:\/\/SwiftErrorKit\/documentation\/SwiftErrorKit\/TagErrorAttribute","abstract":[{"type":"text","text":"An error attribute to assign a tag to an error."}],"navigatorTitle":[{"text":"TagErrorAttribute","kind":"identifier"}],"url":"\/documentation\/swifterrorkit\/tagerrorattribute","title":"TagErrorAttribute","role":"symbol","fragments":[{"text":"struct","kind":"keyword"},{"text":" ","kind":"text"},{"kind":"identifier","text":"TagErrorAttribute"}],"type":"topic"},"doc://SwiftErrorKit/documentation/SwiftErrorKit/SourceErrorAttribute/input":{"title":"SourceErrorAttribute.input","kind":"symbol","identifier":"doc:\/\/SwiftErrorKit\/documentation\/SwiftErrorKit\/SourceErrorAttribute\/input","fragments":[{"text":"case","kind":"keyword"},{"kind":"text","text":" "},{"kind":"identifier","text":"input"}],"abstract":[{"type":"text","text":"Error resulting from user interaction or input data problem."},{"type":"text","text":" "},{"type":"text","text":"Invalid input data, form submission error, user request error, …"}],"type":"topic","role":"symbol","url":"\/documentation\/swifterrorkit\/sourceerrorattribute\/input"},"doc://SwiftErrorKit/documentation/SwiftErrorKit/SourceErrorAttribute/system":{"url":"\/documentation\/swifterrorkit\/sourceerrorattribute\/system","role":"symbol","fragments":[{"text":"case","kind":"keyword"},{"text":" ","kind":"text"},{"text":"system","kind":"identifier"}],"title":"SourceErrorAttribute.system","identifier":"doc:\/\/SwiftErrorKit\/documentation\/SwiftErrorKit\/SourceErrorAttribute\/system","kind":"symbol","type":"topic","abstract":[{"type":"text","text":"Error caused by a system that the process runs on."},{"type":"text","text":" "},{"type":"text","text":"Hardware failure, OS-level error, resource exhaustion,"},{"type":"text","text":" "},{"type":"text","text":"environment configuration, …"}]},"doc://SwiftErrorKit/SQ":{"identifier":"doc:\/\/SwiftErrorKit\/SQ","title":"Swift.Equatable","type":"unresolvable"},"doc://SwiftErrorKit/documentation/SwiftErrorKit/SourceErrorAttribute/ErrorAttribute-Implementations":{"abstract":[],"identifier":"doc:\/\/SwiftErrorKit\/documentation\/SwiftErrorKit\/SourceErrorAttribute\/ErrorAttribute-Implementations","kind":"article","title":"ErrorAttribute Implementations","type":"topic","role":"collectionGroup","url":"\/documentation\/swifterrorkit\/sourceerrorattribute\/errorattribute-implementations"},"doc://SwiftErrorKit/documentation/SwiftErrorKit/SeverityErrorAttribute":{"title":"SeverityErrorAttribute","url":"\/documentation\/swifterrorkit\/severityerrorattribute","fragments":[{"kind":"keyword","text":"enum"},{"text":" ","kind":"text"},{"kind":"identifier","text":"SeverityErrorAttribute"}],"type":"topic","identifier":"doc:\/\/SwiftErrorKit\/documentation\/SwiftErrorKit\/SeverityErrorAttribute","navigatorTitle":[{"kind":"identifier","text":"SeverityErrorAttribute"}],"kind":"symbol","abstract":[{"text":"An error attribute to express error severity in your system.","type":"text"}],"role":"symbol"},"doc://SwiftErrorKit/documentation/SwiftErrorKit/TraceErrorAttribute":{"identifier":"doc:\/\/SwiftErrorKit\/documentation\/SwiftErrorKit\/TraceErrorAttribute","navigatorTitle":[{"kind":"identifier","text":"TraceErrorAttribute"}],"abstract":[{"text":"An error attribute to track a source location.","type":"text"}],"fragments":[{"text":"struct","kind":"keyword"},{"text":" ","kind":"text"},{"text":"TraceErrorAttribute","kind":"identifier"}],"title":"TraceErrorAttribute","url":"\/documentation\/swifterrorkit\/traceerrorattribute","kind":"symbol","role":"symbol","type":"topic"},"doc://SwiftErrorKit/documentation/SwiftErrorKit/RankErrorAttribute":{"abstract":[{"type":"text","text":"An error attribute to assign a tracked issue a priority."}],"identifier":"doc:\/\/SwiftErrorKit\/documentation\/SwiftErrorKit\/RankErrorAttribute","kind":"symbol","navigatorTitle":[{"text":"RankErrorAttribute","kind":"identifier"}],"url":"\/documentation\/swifterrorkit\/rankerrorattribute","title":"RankErrorAttribute","role":"symbol","fragments":[{"kind":"keyword","text":"enum"},{"text":" ","kind":"text"},{"text":"RankErrorAttribute","kind":"identifier"}],"type":"topic"},"doc://SwiftErrorKit/documentation/SwiftErrorKit/SourceErrorAttribute/valueDescription":{"fragments":[{"kind":"keyword","text":"var"},{"kind":"text","text":" "},{"kind":"identifier","text":"valueDescription"},{"text":": [","kind":"text"},{"kind":"typeIdentifier","preciseIdentifier":"s:SS","text":"String"},{"text":"]","kind":"text"}],"identifier":"doc:\/\/SwiftErrorKit\/documentation\/SwiftErrorKit\/SourceErrorAttribute\/valueDescription","type":"topic","url":"\/documentation\/swifterrorkit\/sourceerrorattribute\/valuedescription","abstract":[],"kind":"symbol","role":"symbol","title":"valueDescription"},"doc://SwiftErrorKit/documentation/SwiftErrorKit":{"title":"SwiftErrorKit","url":"\/documentation\/swifterrorkit","type":"topic","identifier":"doc:\/\/SwiftErrorKit\/documentation\/SwiftErrorKit","kind":"symbol","abstract":[{"text":"Error attribution and commons.","type":"text"}],"role":"collection"}}}